tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -108,6 +115,22 @@ def teardown(self):
         self.command_queue = None
         self.browser = None
 
+    def switch_executor(self, executor_implementation):
+        assert self.executor is not None
```

Comment:
True, and that is the reason why `assert` is used instead of an if statement, right? Or did I miss your point?

A similar use of `assert` below:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/wpt_tools/wpt/tools/wptrunner/wptrunner/testrunner.py;l=555

Reply:
Sorry, what I meant was that mypy (which this repo has) can verify the non-nullibility of `TestRunner.executor` statically, whereas the `TestRunnerManager` state can't be.